test/fast/deploy/public/index-test.js

```diff
@@ -1,59 +1,96 @@
-var test = require('tape')
-var proxyquire = require('proxyquire')
-var mkdir = require('mkdirp')
-var fs = require('fs')
-var sinon = require('sinon')
+let fs = require('fs')
+let mkdir = require('mkdirp')
+let proxyquire = require('proxyquire')
+let sinon = require('sinon')
+let test = require('tape')
 let mkdirStub = sinon.stub(mkdir, 'sync')
-let copyStub = sinon.stub()
+let publishStub = sinon.stub().callsFake(({Bucket, fingerprint, ignore, prune}, callback) => {callback()})
 
-var deployPublic = proxyquire('../../../../src/deploy/public', {
-  './_copy-to-s3': copyStub
-})
+let deployPublic = proxyquire('../../../../src/deploy/public', {'./_publish-to-s3': publishStub})
 
-var base = {
-  appname: ['testapp']
+let arc = {
+  appname: ['testapp'],
+  static: [
+    ['staging', 'stagingbucket'],
+    ['production', 'prodbucket']
+  ]
 }
 
-test('deploy/public should ignore if arc is missing static param', t=> {
+test('Deploy/public should bail if arc is missing @static pragma', t=> {
   t.plan(1)
-  deployPublic({ arc: base, env: '' }, () => {
+  deployPublic({ arc: {appname: ['testapp']}, env: '' }, () => {
     t.ok(!mkdirStub.called, 'mkdir not called')
     t.end()
   })
 })
 
-test('deploy/public should make a public dir if theres a static param', t=> {
+test('Deploy/public should make a public dir if arc specifies @static pragma', t=> {
   t.plan(1)
-  var arc = Object.assign(base, {static:[['staging', 'stagingbucket'], ['production', 'prodbucket']]})
-  let readStub = sinon.stub(fs, 'readdir').callsFake((path, callback) => callback(null, []))
   deployPublic({ arc, env: '' }, () => {
     t.ok(mkdirStub.called, 'mkdir called')
-    readStub.restore()
+    mkdirStub.resetHistory()
     t.end()
   })
 })
 
-test('deploy/public should not invoke copy if public dir is empty', t=> {
-  t.plan(1)
-  var arc = Object.assign(base, {static:[['staging', 'stagingbucket'], ['production', 'prodbucket']]})
-  let readStub = sinon.stub(fs, 'readdir').callsFake((path, callback) => callback(null, []))
+test('Fingerprinting should be disabled by default', t=> {
```

Comment:
Just a suggestion, not at all required, but I'd probably split this into three separate test cases.

Reply:
Fair! I'm pretty exhausted atm so I'll get this up in master, feel free to break this and orphan / pruning tests down further if you like – no worries if not, ofc!